subsys/shell/shell.c

```diff
@@ -1231,6 +1241,11 @@ void shell_thread(void *shell_handle, void *dummy1, void *dummy2)
 		return;
 	}
 
+	if ((bool)log_backend && IS_ENABLED(CONFIG_LOG)) {
+		shell_log_backend_enable(shell->log_backend, (void *)shell,
+					 (bool)log_level);
```

Comment:
I do not understand this conversion. `log_level` is a pointer, you are converting it to `bool` and passing to function expecting `u32_t` ?

Reply:
fixed. I also refactored a bit to cast early those `void *` to local variables.